pkg/provider/cloud_test.go

```diff
@@ -58,41 +62,45 @@ func makeCloudConfig(kubeconfig, namespace string, loadbalancerEnabled, instance
-		{allConf, makeCloudConfig(kubeconfig, ns, false, false, 5), nil},
-	}
+var _ = Describe("Cloud config", func() {
 
-	for _, test := range tests {
-		config, err := NewCloudConfigFromBytes([]byte(test.configBytes))
-		if !reflect.DeepEqual(config, test.expectedCloudConfig) {
-			t.Errorf("Expected: %v, got %v", test.expectedCloudConfig, config)
-		}
-		if test.expectedError != nil && err != nil && err.Error() != test.expectedError.Error() {
-			t.Errorf("Expected: '%v', got '%v'", test.expectedError, err)
-		}
-	}
-}
+	DescribeTable("Get CloudConfig from Bytes", func(configBytes string, expectedCloudConfig CloudConfig, expectedError error) {
+		config, err := NewCloudConfigFromBytes([]byte(configBytes))
+		Expect(config).To(Equal(expectedCloudConfig))
+		Expect(err).To(BeNil())
+	},
+		Entry("With Minimal Configuration", minimalConf, makeCloudConfig(infraKubeConfigPath, "", true, true, 5), nil),
+		Entry("With LoadBalancer Configuration", loadbalancerConf, makeCloudConfig(infraKubeConfigPath, "", false, true, 3), nil),
+		Entry("With Instance Configuration", instancesConf, makeCloudConfig(infraKubeConfigPath, "", true, false, 5), nil),
+		Entry("With Namespace Configuration", namespaceConf, makeCloudConfig(infraKubeConfigPath, ns, true, true, 5), nil),
+		Entry("With full Configuration", allConf, makeCloudConfig(infraKubeConfigPath, ns, false, false, 5), nil),
+	)
 
-func TestKubevirtCloudProviderFactory(t *testing.T) {
-	// Calling kubevirtCloudProviderFactory without config should return an error
-	_, err := kubevirtCloudProviderFactory(nil)
-	if err == nil {
-		t.Error("Expected: 'No kubevirt cloud provider config file given', got 'nil'")
-	} else if err.Error() != "No kubevirt cloud provider config file given" {
-		t.Errorf("Expected: 'No kubevirt cloud provider config file given', got '%v'", err)
-	}
+	Describe("KubeVirt Cloud Factory", func() {
 
-	_, err = kubevirtCloudProviderFactory(strings.NewReader(invalidKubeconf))
-	if err == nil || !strings.Contains(err.Error(), "couldn't get version/kind; json parse error") {
-		t.Errorf("Expected error containing: 'couldn't get version/kind; json parse error', got '%v'", err)
-	}
-}
+		Context("With nil config", func() {
+
+			It("Should return an error", func() {
+				_, err := kubevirtCloudProviderFactory(nil)
+				Expect(err).To(HaveOccurred())
+				Expect(err.Error()).To(Equal("No kubevirt cloud provider config file given"))
+			})
+
+		})
+
+		Context("With invalid infraKubeConfig", func() {
+
+			It("Should return an error", func() {
+				infraKubeConfig, err := ioutil.TempFile("", "infraKubeConfig")
+				Expect(err).NotTo(HaveOccurred())
+				infraKubeConfig.Write([]byte(invalidKubeconf))
+				_, err = kubevirtCloudProviderFactory(strings.NewReader(fmt.Sprintf("kubeconfig: %s", infraKubeConfig.Name())))
+				Expect(err).To(HaveOccurred())
+				Expect(strings.Contains(err.Error(), "couldn't get version/kind; json parse error")).To(BeTrue())
+				os.Remove(infraKubeConfig.Name())
```

Comment:
If the test will fail becuase of some reasons (like code change) then `os.Remove` won't be executed.
Since you created a context with invalid infraKubeConfig then it's ok to implement `BeforeEach` and `AfterEach` functions to initialize and cleanup in there.